### PR TITLE
[SW-823] Convert NetworkForm component to Composition API and TypeScript

### PR DESCRIPTION
**Pull request by pstachel**

### CODE REVIEW

File: src/types/index.ts [hunk 1]
```diff
@@ -188,17 +188,20 @@ export interface IRawMultisigAccount {
...
-  backendUrl: string
+export interface INetworkBase {
+  url: string
+  name: string
+  middlewareUrl: string
   compilerUrl: string
+  backendUrl: string
+}
+
+export interface INetwork extends INetworkBase{
```
Comment (peronczyk) [inline src/types/index.ts:199]: Missing space at the end :)

File: src/popup/pages/NetworkForm.vue [hunk 2]
```diff
@@ -46,112 +46,140 @@
...
-        ...NETWORK_PROPS,
+  setup(props, { root }) {
+    const networks = useGetter('networks');
+    const selectNetwork = useDispatch('selectNetwork');
+
+    const newNetwork = ref<INetworkForm>({
+      ...NETWORK_PROPS,
+    });
+
+    const formConfig = [
```
Comment (peronczyk) [inline src/popup/pages/NetworkForm.vue:87]: I suggest to add types here, mostly to bind the `key` with the `INetworkBase` properties.

File: src/popup/pages/NetworkForm.vue [hunk 3]
```diff
@@ -46,112 +46,140 @@
...
+
 import BtnMain from '../components/buttons/BtnMain.vue';
 import InputField from '../components/InputField.vue';
 import PlusCircleIcon from '../../icons/plus-circle.svg?vue-component';
 
+interface INetworkForm extends INetworkBase {
+  index?: number
+}
+
 export const NETWORK_PROPS = {
```
Comment (peronczyk) [inline src/popup/pages/NetworkForm.vue:66]: I suggest to add type here.

File: src/popup/pages/NetworkForm.vue [hunk 4]
```diff
@@ -46,112 +46,140 @@
...
 import { defaultNetwork } from '../utils';
+import { useDispatch, useGetter } from '../../composables/vuex';
+import type { INetworkBase } from '../../types';
+
 import BtnMain from '../components/buttons/BtnMain.vue';
 import InputField from '../components/InputField.vue';
 import PlusCircleIcon from '../../icons/plus-circle.svg?vue-component';
 
+interface INetworkForm extends INetworkBase {
+  index?: number
```
Comment (peronczyk) [inline src/popup/pages/NetworkForm.vue:63]: This property is weird in my opinion :) On my branch I'm removing it totally. Here I suggest just to add this to the INetworkBase as optional maybe.

**Review by peronczyk — APPROVED** [review]

File: src/popup/pages/NetworkForm.vue [hunk 5]
```diff
@@ -46,112 +46,145 @@
...
-      PlusCircleIcon,
-      newNetwork: {
-        ...NETWORK_PROPS,
+  setup(props, { root }) {
+    const networks = useGetter('networks');
+    const selectNetwork = useDispatch('selectNetwork');
+
+    const newNetwork = ref<INetworkBase>({
+      ...NETWORK_PROPS,
+    });
```
Comment (konradodo) [inline src/popup/pages/NetworkForm.vue:90]:
```suggestion
    const newNetwork = ref<INetworkBase>(NETWORK_PROPS);
```

**Review by konradodo — APPROVED** [review]

**Pull request merged**